### DISCUSSION TITLE
Guests heading for ATM won't withdraw cash upon arrival

### DISCUSSION THREAD

**Discussion by astraylife1** [OpenRCT2/OpenRCT2#22158] [Feature requests]
### Operating System

MacOS

### OpenRCT2 build

OpenRCT2, v0.4.11 (71a8eee on develop)

### Base game

RollerCoaster Tycoon 2

### Describe the issue

Guests heading for the ATM will arrive and not withdraw cash and walk away, still complaining that they are running out of cash. They may attempt several times to withdraw cash. Sometimes they are successful but many times they are not.

### Steps to reproduce

Add and ATM to Forest Frontiers
Set guest cash ins scenario options to $1
Guests enter park, realize they are low on cash and head for ATM
Fail to withdraw cash

**Comment by spacek531** [comment]
I can confirm this happens on OpenRCT2, v0.4.11-199-gd4f97d3 (d4f97d3 on develop) provided by GitHub

I noticed that the first couple of guests were able to get cash from the cash machine, but later guests almost never could, but infrequently some would withdraw cash successfully.

**Comment by spacek531** [comment]
https://github.com/OpenRCT2/OpenRCT2/blob/71a8eee5c652b87b5e52c4f10e6c841bcf8987f2/src/openrct2/entity/Guest.cpp#L3310-L3311

It appears there is a random chance the guest won't withdraw cash. The odds of the guest withdrawing cash increase with guest happiness. The game is working as intended, although it would be more intuitive to put this check when the guest decides if they are heading to a cash machine.

**Comment by astraylife1** [comment]
I have guests with full happiness try multiple times to head to an ATM and repeatedly fail. Doesn’t seem random.

**Comment by spacek531** [comment]
They have to have at least 80 energy too.

**Comment by leicestersquare** [comment]
"Spending more money in this Park? I have to think about it." As a guest thought would be a way of communicating that behaveviour to the player without changing it.

**Comment by astraylife1** [comment]
It seems odd that this behavior exists as the guests don’t do this when they are heading for a food, drink, or toilet stall.

**Comment by leicestersquare** [comment]
I’m not sure how much that would throw off the balance of the economy. Having a guest withdraw money is essentially a fresh guest entering the park.

The behavior is known by the community to a point: https://m.youtube.com/watch?v=6aYRv-H6OXc

I get you though. It feels a little random. …well it is after all.

**Comment by astraylife1** [comment]
Hmm… seems counterintuitive that an unhappy guest wouldn’t withdraw cash to fulfill their needs to eat/drink/ride if they are already heading to the cash machine. Seems like the check should happen to decide whether or not they even go to the cash machine. Because now the guests are walking around heading for a cash machine and not riding rides which would increase their happiness. 🤦‍♂️
